cmd/carbonapi/main.go

```diff
@@ -151,6 +154,17 @@ func main() {
 
 				wg.Done()
 			}(listener, isTLS)
+
+			go func() {
+				stop := make(chan os.Signal, 1)
+				signal.Notify(stop, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
Don't set signal handler in a loop. More better - set one handler and shutdown all listeners in them.